### PR TITLE
Anomali threat stream v3

### PR DESCRIPTION
**Pull request by ilappe**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30158

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py [hunk 1]
```diff
@@ -0,0 +1,1239 @@
...
+            readable_output=readable_output,
+            raw_response=report_info,
+        )
+    else:
+        return status, verdict
+
+
+def file_name_to_valid_string(file_name):
+    try:
+        # In case the user uses Cortex XSOAR version < 5.0 and the new docker image will not be automatically changed
```
Comment (yaakovi) [inline Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py:1024]: remove as this comment is not relevant anymore

File: Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py [hunk 2]
```diff
@@ -0,0 +1,1239 @@
...
+    return results
+
+
+def get_ip_reputation(client: Client, score_calc: DBotScoreCalculator, ip, status, threshold=None):
+    """
+        Checks the reputation of given ip from ThreatStream and
+        returns the indicator with highest confidence score.
+    """
+    # get the indicator
+    params = dict(value=ip, type=DBotScoreType.IP, status=status, limit=0)
```
Comment (yaakovi) [inline Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py:442]: why are you initializing it like that?

File: Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py [hunk 3]
```diff
@@ -0,0 +1,1239 @@
...
+def get_indicators(client: Client, **kwargs):
+    """
+        Returns filtered indicators by parameters from ThreatStream.
+        By default the limit of indicators result is set to 20.
+    """
+    limit = kwargs['limit'] = int(kwargs.get('limit', 20))
+    offset = kwargs['offset'] = 0
+    url = "v2/intelligence/"
+    if 'query' in kwargs:
+        url += f"?{kwargs.pop('query')}"
```
[the remark this answers is not in the document]
Comment (ilappe) [reply Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py:1133]: because the value of the query should be the Anomaly Query language which is in the form of k1=v1&k2=v2

File: Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py [hunk 4]
```diff
@@ -1057,7 +1049,7 @@ def submit_report(client: Client, submission_type, submission_value, submission_
             raise DemistoException(f'{THREAT_STREAM} - Entry {submission_value} does not contain a file.')
 
         uploaded_file = open(file_info['path'], 'rb')
-        file_name = file_name_to_valid_string(file_info.get('name'))
+        file_name = file_info.get('name')
```
Comment (yaakovi) [inline Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.py:1052]: don't you need the `emoji` handling? why did you remove it?

File: Packs/Anomali_ThreatStream/ReleaseNotes/2_0_0.md [hunk 5]
```diff
@@ -1,5 +1,4 @@
 
 #### Integrations
 ##### New: Anomali ThreatStream v3
-- Use Anomali ThreatStream to query and submit threats. (Available from Cortex XSOAR 5.0.0).
-- Updated the Docker image to: *demisto/emoji:1.0.0.24037*
\ No newline at end of file
+- Use Anomali ThreatStream to query and submit threats. (Available from Cortex XSOAR 5.0.0).
```
Comment (yaakovi) [inline Packs/Anomali_ThreatStream/ReleaseNotes/2_0_0.md:4]:
```suggestion
- Use Anomali ThreatStream to query and submit threats. (Available from Cortex XSOAR 6.0.0).
```

File: Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.yml [hunk 6]
```diff
@@ -0,0 +1,1658 @@
...
+  isfetch: false
+  longRunning: false
+  longRunningPort: false
+  runonce: false
+  script: '-'
+  subtype: python3
+  type: python
+tests:
+- ThreatStream-Test
+fromversion: 5.0.0
```
Comment (yaakovi) [inline Packs/Anomali_ThreatStream/Integrations/AnomaliThreatStreamv3/AnomaliThreatStreamv3.yml:1658]:
```suggestion
fromversion: 6.0.0
```

**Review by yaakovi — APPROVED** [review]

**Review by yaakovi — APPROVED** [review]

**Comment by yaakovi** [comment]
failed on irrelevant tests.
